File: LeetCode263/LeetCode263.java
class Solution {
    private int devideFactor(int num, int factor)
    {
        while (num % factor == 0)
        {
            num /= factor;
        }
    
        return num;
    }
    
    public boolean isUgly(int num) 
    {
        if (num <= 0)
            return false;

        num = devideFactor(num, 2);
        num = devideFactor(num, 3);
        num = devideFactor(num, 5);

        if (num == 1)
            return true;
        else
            return false;
    }
}

File: LeetCode263/LeetCode263.py
class Solution:
    def isUgly(self, num: int) -> bool:
        if num <= 0:
            return False

        num = self.devideFactor(num, 2)
        num = self.devideFactor(num, 3)
        num = self.devideFactor(num, 5)

        if num == 1:
            return True
        else:
            return False
        
    def devideFactor(self, num: int, factor: int) -> int:
        while num % factor == 0:
            num /= factor
        
        return num